src/main/kotlin/net/ccbluex/liquidbounce/features/command/commands/client/fakeplayer/FakePlayer.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * This file is part of LiquidBounce (https://github.com/CCBlueX/LiquidBounce)
+ *
+ * Copyright (c) 2015 - 2024 CCBlueX
+ *
+ * LiquidBounce is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * LiquidBounce is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with LiquidBounce. If not, see <https://www.gnu.org/licenses/>.
+ */
+package net.ccbluex.liquidbounce.features.command.commands.client.fakeplayer
+
+import com.mojang.authlib.GameProfile
+import net.ccbluex.liquidbounce.interfaces.LivingEntityAddition
+import net.ccbluex.liquidbounce.interfaces.OtherClientPlayerEntityAddition
+import net.minecraft.client.network.OtherClientPlayerEntity
+import net.minecraft.client.world.ClientWorld
+import net.minecraft.entity.damage.DamageSource
+import java.util.function.Consumer
+
+/**
+ * This class represents a Fake Player implementing
+ * attackability and assured totem pops instead of death
+ * into [OtherClientPlayerEntity].
+ */
+open class FakePlayer(
+    clientWorld: ClientWorld?,
+    gameProfile: GameProfile?,
+    private val remover: Consumer<FakePlayer>
```

Comment:
This function is too specifiy. Maybe change it to a `() -> Unit` function and call it `onRemoval`